packages/clerk-js/src/ui/components/KeylessPrompt/index.tsx

```diff
@@ -252,111 +273,106 @@ const _KeylessPrompt = (_props: KeylessPromptProps) => {
-              >
-                <path
-                  d='M3.75 8H12.25'
-                  stroke='currentColor'
-                  strokeWidth='1.5'
-                  strokeLinecap='round'
-                  strokeLinejoin='round'
-                />
-              </svg>
-            </button>
-          )}
-        </Flex>
-
-        {isExpanded && (
-          <p
+          <svg
+            width='1rem'
+            height='1rem'
+            viewBox='0 0 16 16'
+            fill='none'
+            aria-hidden={!isExpanded || claimed}
+            aria-label='Minimize keyless overlay'
+            xmlns='http://www.w3.org/2000/svg'
             css={css`
-              color: #b4b4b4;
-              font-size: 0.75rem;
-              font-weight: 400;
-              line-height: 1rem;
-              max-width: 14.625rem;
-              min-height: 2rem;
-              animation: show-description 260ms ease-out forwards;
+              color: #8c8c8c;
+              transition: color 130ms ease-out;
+              visibility: ${isExpanded && !claimed ? 'visible' : 'hidden'};
+              :hover {
+                color: #eeeeee;
+              }
+              animation: show-button 200ms cubic-bezier(0.4, 0, 0, 1.1) forwards;
 
-              @keyframes show-description {
+              @keyframes show-button {
                 from {
-                  transform: translateY(-1.8px);
+                  transform: scaleX(0.9);
                   opacity: 0;
                 }
                 to {
-                  transform: translateY(0);
+                  transform: scaleX(1);
                   opacity: 1;
                 }
               }
             `}
           >
-            {claimed ? (
-              <>
-                You claimed this application, but haven&apos;t set keys in your environment. Get your keys from the
-                Clerk Dashboard.
-              </>
-            ) : (
-              <>
-                API keys were missing so we generated them for you. Link this instance to your Clerk account to make
-                configuration changes.{' '}
-                <Link
-                  href='https://clerk.com/docs/keyless'
-                  sx={t => ({
-                    color: t.colors.$whiteAlpha600,
-                    textDecoration: 'underline solid',
-                    transition: `${t.transitionTiming.$common} ${t.transitionDuration.$fast}`,
-                    ':hover': {
-                      color: t.colors.$whiteAlpha800,
-                    },
-                  })}
-                >
-                  Learn more
-                </Link>
-              </>
-            )}
-          </p>
+            <path
+              d='M3.75 8H12.25'
+              stroke='currentColor'
+              strokeWidth='1.5'
+              strokeLinecap='round'
+              strokeLinejoin='round'
+            />
+          </svg>
+        </button>
+
+        {isExpanded && (
```

Comment:
This `<div>` must always be in the DOM so this conditional here must be removed. By toggling the `hidden` attribute (which we already do) the functionality should be correct already.

Reply:
Good catch!